### PR TITLE
implement numeric validation

### LINKED ISSUE #52: Add numeric validation rule
### Description
Add a `numeric` validation rule that ensures a string contains only numeric characters (0-9).

### Usage Example
```go
type Product struct {
    // +govalid:numeric
    SKU string 
}
```

### Expected Behavior
- Valid: "123456", "000", "9"
- Invalid: "123abc", "12.34", "12-34", " 123", ""

### Implementation Notes
- Should validate that the string contains only digits 0-9
- No spaces, decimal points, or other characters allowed
- Empty strings should be invalid
- Similar to go-playground/validator's `numeric` tag

### Rationale
This is a commonly needed validation for fields like:
- Product codes/SKUs
- Numeric IDs stored as strings
- Phone numbers (without formatting)
- Postal codes (in some countries)

### PR DESCRIPTION
**Pull request by ferenc-zagon**
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [x] Code passes `make golangci-lint` without errors
- [x] All tests pass: `make test`
- [x] Code follows existing patterns and conventions
- [x] No hard-coded values (use constants or configuration)
- [x] Error messages are clear and actionable

### Testing Requirements
- [x] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [x] Unit tests implemented with boundary value testing (`test/unit/`)
- [x] Benchmark tests comparing against popular validation libraries (`test/benchmark/`)
  - [x] [go-playground/validator](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)
  - [x] [asaskevich/govalidator](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)
  - [x] [go-ozzo/ozzo-validation](https://github.com/go-ozzo/ozzo-validation) (BenchmarkOzzoValidation*)
  - [x] [gookit/validate](https://github.com/gookit/validate) (BenchmarkGookitValidate*)
- [x] All test files generated: `cd test && go generate`

### Performance Standards
- [x] Zero allocations in validation logic (verified via benchmarks)
- [x] Performance improvement over existing validation libraries (minimum 2x faster)
- [x] Benchmark results added to `test/benchmark/README.md`

### Documentation
- [x] README.md updated with new marker documentation
- [x] Code includes appropriate comments and examples
- [x] Validator usage examples provided

### Integration
- [x] Validator scaffold generated with `make generate-validator MARKER=yourmarker`
- [x] Registry files automatically updated
- [x] Binary rebuilt and installed: `go install ./cmd/govalid/`

### Final Verification
- [x] All GitHub Actions CI checks pass
- [x] No breaking changes to existing functionality
- [x] Contribution follows project license requirements

## Related Issues
Closes #52

### CODE REVIEW

File: validation/validationhelper/numeric.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	}
+
+	for _, ch := range s {
+		if ch < '0' || ch > '9' {
+			return false
+		}
+	}
+
+	return true
+}
```
Comment (ccoVeille) [inline validation/validationhelper/numeric.go:17]: So -42 is not numeric ?
Same for 3.14 ?

alphanumeric, that is commonly accepted and understood being [A-Za-z0-9]

But numeric sounds like numbers. And number can be float, negative...

So maybe "numeric" is not the best name

It might be OK but I'm raising the point just in case

**Comment by BlackBuck** [comment]
> # Related Issues

@ferenc-zagon please add the related issue number in this section

File: validation/validationhelper/numeric.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+		return false
+	}
+
+	for _, ch := range s {
+		if ch < '0' || ch > '9' {
+			return false
+		}
+	}
+
+	return true
```
Comment (BlackBuck) [inline validation/validationhelper/numeric.go:16]: You could change it to the following (since other markers follow similar procedure):

```go
  for _, ch := range s {
		  if ch < '0' || ch > '9' {
			  return false
		  }
	  }
  
  return s != ""
```

Comment (ferenc-zagon) [reply validation/validationhelper/numeric.go]: Good idea, thanks!

**Review by sivchari — APPROVED** [review]
LGTM
@taua-almeida @BlackBuck 
Please review it if you have time 🙏
